src/advert_phone_numbers.py

```diff
@@ -10,29 +10,31 @@
 class AdvertPhoneNumbers(object):
     @staticmethod
     def generate(data, phone_number_uuid_table, advert_phone_numbers_csv_output_path):
-        advert_phone_numbers = set()
-
         '''
         Generates a csv file with normalised phone numbers for respondents who sent messages 
         that were not labelled as Noise_Other_Project.
 
-        :param data:TracedData objects that have been manually labelled. 
-        :type: TracedData
+        :param data: TracedData objects that have been manually labelled. 
+        :type: List of TracedData
         :param: phone_number_uuid_table
-        :type: a look up table containing uuids to retrieve phone numbers from.
-        :return: advert_phone_numbers_csv_output_path
+        :type: a coredatamodules.PhoneNumberUuidTable look up table containing uuids to retrieve phone numbers from.
+        :return: advert_phone_numbers
         :rtype: csv file
         '''
+
+        advert_phone_numbers = set()
+        
         for td in data:
             for plan in PipelineConfiguration.RQA_CODING_PLANS:
-                    if (plan.binary_coded_field is not None and td[plan.binary_coded_field]["CodeID"] != "code-NOP-4eb70633") \
-                        or td[plan.coded_field][0]["CodeID"] != "code-NOP-4eb70633":
-                        advert_phone_numbers.add(phone_number_uuid_table.get_phone(td['uid']))
-        
+                if plan.binary_code_scheme is not None and td[plan.binary_coded_field]["CodeID"] not in ["code-NOP-4eb70633", "code-STOP-08b832a8", "code-NA-f93d3eb7","code-NR-5e3eee23"]:
```

Comment:
We should extract this list into a constant

Reply:
List moved to pipeline_configuration.py as constant https://github.com/AfricasVoices/Project-ICRAF/pull/16/commits/95c32388bf14c357e30a904ae3ee78dabf60587e.